embeddings/hyperparameter_search/configspace.py

```diff
@@ -102,3 +103,28 @@ def _retrieve_embedding_type(embedding_name: str) -> str:
         else:
             raise TypeError("Embedding type not recognized")
         return embedding_type
+
+    @staticmethod
+    def _parse_yaml_params(parameters: Dict[str, Any]) -> Dict[str, Parameter]:
+        parsed_parameters = {}
+        for param_key, param_values in parameters.items():
+            param_type = param_values.pop("param_type")
+            param: Parameter
+            if param_type == "constant":
+                param = ConstantParameter(**param_values)
+            elif param_type == "searchable":
+                param = SearchableParameter(**param_values)
+            else:
+                raise ValueError("Unrecognized parameter type")
+            parsed_parameters.update({param_key: param})
```

Comment:
consider extracting allowed types definitions outside this method plus maybe good pattern here would be early exit 
```
ALLOWED_PARAMETER_TYPES = {"constant", "searchable"}
def _parse_yaml_params(parameters: Dict[str, Any]) -> Dict[str, Parameter]:
    parsed_parameters = {}
    for param_key, param_values in parameters.items():
        param_type = param_values.get("param_type") # or param_values["param_type"]
        if param_type not in ALLOWED_PARAMETER_TYPES:
            raise ValueError("Unrecognized parameter type")  # early exit
        parameter_parser_type = ConstantParameter if "constant" == param_type else SearchableParameter
        parsed_parameters[param_key] = parameter_parser_type(**param_values)
    return parsed_parameters
```

Reply:
meh, this is not a computationally expensive piece of code

Reply:
I prefer the first version because of its readability. Easier to ask for forgiveness than permission

Reply:
Sure, however main aim of these changes was not to speedup things but to make it either more extensible or hopefully more readable for other developers in the future - for example early exit is common pattern, ALLOWED_PARAMETER_TYPES gives idea that there is certain group of types to expect and there's limit to elifs etc.
Anyway just suggestions and likely a matter of style as well